--- src/tap/Makefile	2015-01-19 20:19:23.000000000 -0500
+++ src/tap/Makefile	2015-01-17 14:37:55.000000000 -0500
@@ -21,11 +21,11 @@
 
 INCLUDE = -I.. -I/System/Library/Frameworks/Kernel.framework/Headers
 CFLAGS = -Wall -mkernel -force_cpusubtype_ALL \
-	-fno-builtin -fno-stack-protector -arch i386 -arch x86_64 \
+	-fno-builtin -fno-stack-protector $(TUNTAP_ARCHS) \
 	-DKERNEL -D__APPLE__ -DKERNEL_PRIVATE -DTUNTAP_VERSION=\"$(TUNTAP_VERSION)\" \
 	-DTAP_KEXT_VERSION=\"$(TAP_KEXT_VERSION)\"
 CCFLAGS = $(CFLAGS)
-LDFLAGS = -Wall -mkernel -nostdlib -r -lcc_kext -arch i386 -arch x86_64 -Xlinker -kext
+LDFLAGS = -Wall -mkernel -nostdlib -r -lcc_kext $(TUNTAP_ARCHS) -Xlinker -kext
 
 CCP = g++
 CC = gcc
diff -u -r -x .svn -x .DS_Store tuntap-20111101/tuntap/src/tun/Makefile tuntap-20111101-patched/tuntap/src/tun/Makefile
--- src/tun/Makefile	2015-01-19 20:19:23.000000000 -0500
+++ src/tun/Makefile	2015-01-17 14:37:53.000000000 -0500
@@ -22,11 +22,11 @@
 
 INCLUDE = -I.. -I/System/Library/Frameworks/Kernel.framework/Headers
 CFLAGS = -Wall -mkernel -force_cpusubtype_ALL \
-	-fno-builtin -fno-stack-protector -arch i386 -arch x86_64 \
+	-fno-builtin -fno-stack-protector $(TUNTAP_ARCHS) \
 	-DKERNEL -D__APPLE__ -DKERNEL_PRIVATE -DTUNTAP_VERSION=\"$(TUNTAP_VERSION)\" \
 	-DTUN_KEXT_VERSION=\"$(TUN_KEXT_VERSION)\"
 CCFLAGS = $(CFLAGS)
-LDFLAGS = -Wall -mkernel -nostdlib -r -lcc_kext -arch i386 -arch x86_64 -Xlinker -kext
+LDFLAGS = -Wall -mkernel -nostdlib -r -lcc_kext $(TUNTAP_ARCHS) -Xlinker -kext
 
 CCP = g++
 CC = gcc
