diff -u -r -x .DS_Store openvpn-2.6.0.old/src/openvpn/forward.c openvpn-2.6.0.new/src/openvpn/forward.c
--- openvpn-2.6.0.old/src/openvpn/forward.c	2023-02-07 17:53:58.000000000 -0500
+++ openvpn-2.6.0.new/src/openvpn/forward.c	2023-02-07 17:53:58.000000000 -0500
@@ -902,7 +902,10 @@

     status = link_socket_read(c->c2.link_socket,
                               &c->c2.buf,
-                              &c->c2.from);
+			                  &c->c2.from,
+			                  c->options.ce.xormethod,
+			                  c->options.ce.xormask,
+			                  c->options.ce.xormasklen);

     if (socket_connection_reset(c->c2.link_socket, status))
     {
@@ -1792,7 +1795,10 @@
                 else
                 {
                     size = link_socket_write(c->c2.link_socket, &c->c2.to_link,
-                                             to_addr);
+                                             to_addr,
+                                             c->options.ce.xormethod,
+                                             c->options.ce.xormask,
+                                             c->options.ce.xormasklen);
                 }

                 /* Undo effect of prepend */
