Index: easy-rsa/2.0/build-dh
===================================================================
--- easy-rsa/2.0/build-dh	(revision 1940)
+++ easy-rsa/2.0/build-dh	(working copy)
@@ -3,8 +3,8 @@
 # Build Diffie-Hellman parameters for the server side
 # of an SSL/TLS connection.
 
-if [ -d $KEY_DIR ] && [ $KEY_SIZE ]; then
-    $OPENSSL dhparam -out ${KEY_DIR}/dh${KEY_SIZE}.pem ${KEY_SIZE}
+if [ -d "$KEY_DIR" ] && [ "$KEY_SIZE" ]; then
+    $OPENSSL dhparam -out "${KEY_DIR}/dh${KEY_SIZE}.pem" "${KEY_SIZE}"
 else
     echo 'Please source the vars script first (i.e. "source ./vars")'
     echo 'Make sure you have edited it to reflect your configuration.'
Index: easy-rsa/2.0/vars
===================================================================
--- easy-rsa/2.0/vars	(revision 1940)
+++ easy-rsa/2.0/vars	(working copy)
@@ -26,7 +26,7 @@
 # This variable should point to
 # the openssl.cnf file included
 # with easy-rsa.
-export KEY_CONFIG=`$EASY_RSA/whichopensslcnf $EASY_RSA`
+export KEY_CONFIG=`"$EASY_RSA/whichopensslcnf" "$EASY_RSA"`
 
 # Edit this variable to point to
 # your soon-to-be-created key
Index: easy-rsa/2.0/whichopensslcnf
===================================================================
--- easy-rsa/2.0/whichopensslcnf	(revision 1950)
+++ easy-rsa/2.0/whichopensslcnf	(working copy)
@@ -3,20 +3,20 @@
 cnf="$1/openssl.cnf"
 
 if [ "$OPENSSL" ]; then
-	if $OPENSSL version | grep -E "0\.9\.6[[:alnum:]]" > /dev/null; then
+	if "$OPENSSL" version | grep -E "0\.9\.6[[:alnum:]]" > /dev/null; then
 		cnf="$1/openssl-0.9.6.cnf"
 	elif $OPENSSL version | grep -E "0\.9\.8[[:alnum:]]" > /dev/null; then
 		cnf="$1/openssl-0.9.8.cnf"
-	elif $OPENSSL version | grep -E "1\.0\.([[:digit:]][[:alnum:]])" > /dev/null; then
+	elif "$OPENSSL" version | grep -E "1\.0\.([[:digit:]][[:alnum:]])" > /dev/null; then
                 cnf="$1/openssl-1.0.0.cnf"
 	else
 		cnf="$1/openssl.cnf"
 	fi
 fi
 
-echo $cnf
+echo "$cnf"
 
-if [ ! -r $cnf ]; then
+if [ ! -r "$cnf" ]; then
     echo "**************************************************************" >&2
     echo "  No $cnf file could be found" >&2
     echo "  Further invocations will fail" >&2
