Index: easy-rsa/2.0/build-dh
===================================================================
--- easy-rsa/2.0/build-dh	(revision 1940)
+++ easy-rsa/2.0/build-dh	(working copy)
@@ -3,8 +3,8 @@
 # Build Diffie-Hellman parameters for the server side
 # of an SSL/TLS connection.
 
-if [ -d $KEY_DIR ] && [ $KEY_SIZE ]; then
-    $OPENSSL dhparam -out ${KEY_DIR}/dh${KEY_SIZE}.pem ${KEY_SIZE}
+if [ -d "$KEY_DIR" ] && [ "$KEY_SIZE" ]; then
+    $OPENSSL dhparam -out "${KEY_DIR}/dh${KEY_SIZE}.pem" "${KEY_SIZE}"
 else
     echo 'Please source the vars script first (i.e. "source ./vars")'
     echo 'Make sure you have edited it to reflect your configuration.'
Index: easy-rsa/2.0/vars
===================================================================
--- easy-rsa/2.0/vars	(revision 1940)
+++ easy-rsa/2.0/vars	(working copy)
@@ -26,7 +26,7 @@
 # This variable should point to
 # the openssl.cnf file included
 # with easy-rsa.
-export KEY_CONFIG=`$EASY_RSA/whichopensslcnf $EASY_RSA`
+export KEY_CONFIG=`"$EASY_RSA/whichopensslcnf" "$EASY_RSA"`
 
 # Edit this variable to point to
 # your soon-to-be-created key
@@ -66,9 +66,3 @@
 export KEY_CITY="SanFrancisco"
 export KEY_ORG="Fort-Funston"
 export KEY_EMAIL="me@myhost.mydomain"
-export KEY_EMAIL=mail@host.domain
-export KEY_CN=changeme
-export KEY_NAME=changeme
-export KEY_OU=changeme
-export PKCS11_MODULE_PATH=changeme
-export PKCS11_PIN=1234
