--- openvpn/socket.c	2012-02-05 06:24:49.000000000 -0500
+++ patched/socket.c	2012-02-29 05:23:32.000000000 -0500
@@ -893,8 +893,13 @@
   else if (flags & SF_USE_IP_PKTINFO)
     {
       int pad = 1;
+#ifndef IPV6_RECVPKTINFO /* Some older Darwin platforms require this */
+      if (setsockopt (sd, IPPROTO_IPV6, IPV6_PKTINFO,
+                     (void*)&pad, sizeof(pad)) < 0)
+#else
       if (setsockopt (sd, IPPROTO_IPV6, IPV6_RECVPKTINFO,
 		      (void*)&pad, sizeof(pad)) < 0)
+#endif
 	msg(M_SOCKERR, "UDP: failed setsockopt for IPV6_RECVPKTINFO");
     }
 #endif
