Index: src/tap/Makefile
===================================================================
--- src/tap/Makefile	(revision 3202)
+++ src/tap/Makefile	(working copy)
@@ -14,7 +14,7 @@
 TAP_KEXT_VERSION = $(TUNTAP_VERSION)
 
 BUNDLE_REGION = English
-BUNDLE_IDENTIFIER = net.sf.tuntaposx.tap
+BUNDLE_IDENTIFIER = net.tunnelblick.tap
 BUNDLE_SIGNATURE = ????
 BUNDLE_PACKAGETYPE = KEXT
 BUNDLE_VERSION = $(TAP_KEXT_VERSION)
Index: src/tun/Makefile
===================================================================
--- src/tun/Makefile	(revision 3202)
+++ src/tun/Makefile	(working copy)
@@ -15,7 +15,7 @@
 TUN_KEXT_VERSION = $(TUNTAP_VERSION)
 
 BUNDLE_REGION = English
-BUNDLE_IDENTIFIER = net.sf.tuntaposx.tun
+BUNDLE_IDENTIFIER = net.tunnelblick.tun
 BUNDLE_SIGNATURE = ????
 BUNDLE_PACKAGETYPE = KEXT
 BUNDLE_VERSION = $(TUN_KEXT_VERSION)
