--- openvpn/src/compat/Makefile.am	2012-06-29 04:09:59.000000000 -0400
+++ patched/src/compat/Makefile.am	2012-07-15 08:24:55.000000000 -0400
@@ -18,6 +18,10 @@
 
 noinst_LTLIBRARIES = libcompat.la
 
+# For Tunnelblick: start of list of files that are compiled directly in the openvpn target
+# (i.e, are specified in src/openvpn/Makefile.am)
+# This comment, and the comment below, are included to try to catch any changes to the list
+# of files used to compile libcompat.
 libcompat_la_SOURCES = \
 	compat.h \
 	compat-stdbool.h \
@@ -27,3 +31,4 @@
 	compat-daemon.c \
 	compat-inet_ntop.c \
 	compat-inet_pton.c
+# For Tunnelblick: end of list of files that are compiled directly in the openvpn target
diff -u -r -x .DS_Store -x '*.svn' openvpn/src/openvpn/Makefile.am patched/src/openvpn/Makefile.am
--- openvpn/src/openvpn/Makefile.am	2012-06-29 04:09:59.000000000 -0400
+++ patched/src/openvpn/Makefile.am	2012-07-14 19:50:29.000000000 -0400
@@ -35,6 +35,14 @@
 sbin_PROGRAMS = openvpn
 
 openvpn_SOURCES = \
+	../compat/compat.h \
+	../compat/compat-stdbool.h \
+	../compat/compat-dirname.c ../compat/compat-dirname.h \
+	../compat/compat-basename.c ../compat/compat-basename.h \
+	../compat/compat-gettimeofday.c ../compat/compat-gettimeofday.h \
+	../compat/compat-daemon.c ../compat/compat-daemon.h \
+	../compat/compat-inet_ntop.c ../compat/compat-inet_ntop.h \
+	../compat/compat-inet_pton.c ../compat/compat-inet_pton.h \
 	base64.c base64.h \
 	basic.h \
 	buffer.c buffer.h \
@@ -113,7 +121,6 @@
 	win32.h win32.c \
 	cryptoapi.h cryptoapi.c
 openvpn_LDADD = \
-	$(top_builddir)/src/compat/libcompat.la \
 	$(SOCKETS_LIBS) \
 	$(OPTIONAL_LZO_LIBS) \
 	$(OPTIONAL_PKCS11_HELPER_LIBS) \
