diff -u -r -x .svn -x .DS_Store Sparkle/SUBasicUpdateDriver.m Sparkle_/SUBasicUpdateDriver.m
--- Sparkle/SUBasicUpdateDriver.m	2008-09-13 23:55:00.000000000 -0400
+++ Sparkle_/SUBasicUpdateDriver.m	2015-10-21 15:26:00.000000000 -0400
@@ -32,7 +32,7 @@
 	[appcast setDelegate:self];
 	NSString *userAgent = [NSString stringWithFormat: @"%@/%@ Sparkle/%@", [aHost name], [aHost displayVersion], ([SPARKLE_BUNDLE objectForInfoDictionaryKey:@"CFBundleVersion"] ?: nil)];
 	NSData * cleanedAgent = [userAgent dataUsingEncoding:NSASCIIStringEncoding allowLossyConversion:YES];
-	userAgent = [NSString stringWithCString:[cleanedAgent bytes] length:[cleanedAgent length]];
+	userAgent = [[[NSString alloc] initWithData: cleanedAgent encoding: NSASCIIStringEncoding] autorelease];
 	[appcast setUserAgentString:userAgent];
 	[appcast fetchAppcastFromURL:URL];
 }
diff -u -r -x .svn -x .DS_Store Sparkle/SUHost.m Sparkle_/SUHost.m
--- Sparkle/SUHost.m	2008-09-13 23:55:00.000000000 -0400
+++ Sparkle_/SUHost.m	2015-10-21 13:54:56.000000000 -0400
@@ -8,7 +8,8 @@
 #import "SUHost.h"
 
 #import "SUSystemProfiler.h"
-#import <sys/mount.h> // For statfs for isRunningOnReadOnlyVolume
+#import <sys/mount.h>  // For statfs for isRunningOnReadOnlyVolume
+#import <sys/sysctl.h> // For sysctlbyname for processorWordSizeString
 
 @implementation SUHost
 
@@ -111,7 +112,7 @@
 	// More likely, we've got a reference to a Resources file by filename:
 	NSString *keyFilename = [self objectForInfoDictionaryKey:SUPublicDSAKeyFileKey];
 	if (!keyFilename) { return nil; }
-	return [NSString stringWithContentsOfFile:[bundle pathForResource:keyFilename ofType:nil]];
+	return [NSString stringWithContentsOfFile:[bundle pathForResource:keyFilename ofType:nil] encoding:NSASCIIStringEncoding error:nil];
 }
 
 - (NSArray *)systemProfile
diff -u -r -x .svn -x .DS_Store Sparkle/Sparkle.xcodeproj/project.pbxproj Sparkle_/Sparkle.xcodeproj/project.pbxproj
--- Sparkle/Sparkle.xcodeproj/project.pbxproj	2008-09-13 23:55:00.000000000 -0400
+++ Sparkle_/Sparkle.xcodeproj/project.pbxproj	2015-10-21 13:54:56.000000000 -0400
@@ -802,7 +802,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = "/usr/bin/env ruby";
-			shellScript = "resources = \"#{ENV[\"BUILT_PRODUCTS_DIR\"]}/#{ENV[\"WRAPPER_NAME\"]}/Resources\"\n`ln -s \"#{resources}/fr.lproj\" \"#{resources}/fr_CA.lproj\"`";
+			shellScript = "resources = \"#{ENV[\"BUILT_PRODUCTS_DIR\"]}/#{ENV[\"WRAPPER_NAME\"]}/Resources\"\n`ln -s \"fr.lproj\" \"#{resources}/fr_CA.lproj\"`";
 		};
 /* End PBXShellScriptBuildPhase section */
 
 