--- Sparkle/Configurations/ConfigCommon.xcconfig	2016-09-07 16:54:16.000000000 -0400
+++ Sparkle_/Configurations/ConfigCommon.xcconfig	2016-09-08 18:45:42.000000000 -0400
@@ -1,14 +1,14 @@
 // Common
 
 SPARKLE_BUNDLE_IDENTIFIER = org.sparkle-project.Sparkle
-SPARKLE_RELAUNCH_TOOL_NAME = Autoupdate
+SPARKLE_RELAUNCH_TOOL_NAME = TunnelblickUpdater
 SPARKLE_FILEOP_TOOL_NAME = fileop
 
 SPARKLE_APPEND_VERSION_NUMBER = 1
 
 // Sparkle usually doesn't allow downgrades as they're usually accidental, but
 // if your app has a downgrade function or URL handler, turn this on
-SPARKLE_AUTOMATED_DOWNGRADES = 0
+SPARKLE_AUTOMATED_DOWNGRADES = 1
 
 // If your app file on disk is named "MyApp 1.1b4", Sparkle usually updates it
 // in place, giving you an app named 1.1b4 that is actually 1.2. Turn the
