diff -u -r -x .svn -x .DS_Store Sparkle-1.5b6-ORIGINAL/SUBasicUpdateDriver.m Sparkle-1.5b6-PATCHED/SUBasicUpdateDriver.m
--- Sparkle-1.5b6-ORIGINAL/SUBasicUpdateDriver.m	2013-06-07 17:50:26.000000000 -0400
+++ Sparkle-1.5b6-PATCHED/SUBasicUpdateDriver.m	2014-07-14 21:24:04.000000000 -0400
@@ -238,6 +238,16 @@
 
 - (void)relaunchHostApp
 {
+	// Give the host an opportunity to SKIP the relaunch
+	if ([[updater delegate] respondsToSelector:@selector(updaterShouldRelaunchApplication:)])
+	{
+		if (  ! [[updater delegate] updaterShouldRelaunchApplication: updater]  ) {
+			[self cleanUp];
+			[self abortUpdate]; // Closes the status window. The update has already happened, so it isn't really an 'abort'
+			return;
+		}
+	}
+	
 	// Give the host app an opportunity to postpone the relaunch.
 	static BOOL postponedOnce = NO;
 	if (!postponedOnce && [[updater delegate] respondsToSelector:@selector(updater:shouldPostponeRelaunchForUpdate:untilInvoking:)])
diff -u -r -x .svn -x .DS_Store Sparkle-1.5b6-ORIGINAL/SUUpdater.h Sparkle-1.5b6-PATCHED/SUUpdater.h
--- Sparkle-1.5b6-ORIGINAL/SUUpdater.h	2013-06-07 17:50:26.000000000 -0400
+++ Sparkle-1.5b6-PATCHED/SUUpdater.h	2014-07-13 11:18:20.000000000 -0400
@@ -87,6 +87,9 @@
 // Sent immediately before installing the specified update.
 - (void)updater:(SUUpdater *)updater willInstallUpdate:(SUAppcastItem *)update;
 
+// Return YES to skip the relaunch (useful for non-application bundles).
+- (BOOL)updaterShouldRelaunchApplication:(SUUpdater *)updater;
+
 // Return YES to delay the relaunch until you do some processing; invoke the given NSInvocation to continue.
 - (BOOL)updater:(SUUpdater *)updater shouldPostponeRelaunchForUpdate:(SUAppcastItem *)update untilInvoking:(NSInvocation *)invocation;
 
diff -u -r -x .svn -x .DS_Store Sparkle-1.5b6-ORIGINAL/SUUpdater.m Sparkle-1.5b6-PATCHED/SUUpdater.m
--- Sparkle-1.5b6-ORIGINAL/SUUpdater.m	2013-06-07 17:50:26.000000000 -0400
+++ Sparkle-1.5b6-PATCHED/SUUpdater.m	2014-07-13 20:43:28.000000000 -0400
@@ -365,6 +365,11 @@
 	[self unregisterAsObserver];
 	[host release];
 	if (checkTimer) { [checkTimer invalidate]; }
+	NSBundle * ourBundle = [self hostBundle];
+	if (  [sharedUpdaters objectForKey: ourBundle]  ) {
+		[sharedUpdaters removeObjectForKey: [NSValue valueWithNonretainedObject: ourBundle]];
+	}
+	
 	[super dealloc];
 }
 
