diff -u -r -x .svn -x .DS_Store -N Sparkle-1.5b6-OldPatches/SUBasicUpdateDriver.m Sparkle-1.5b6-NewPatches/SUBasicUpdateDriver.m
--- Sparkle-1.5b6-OldPatches/SUBasicUpdateDriver.m	2015-01-08 19:47:39.000000000 -0500
+++ Sparkle-1.5b6-NewPatches/SUBasicUpdateDriver.m	2015-01-15 13:04:56.000000000 -0500
@@ -356,7 +356,7 @@
 
 - (void)downloadUpdate
 {
-	download = [[NSURLDownload alloc] initWithRequest:[NSURLRequest requestWithURL:[updateItem fileURL]] delegate:self];	
+	download = [[NSURLDownload alloc] initWithRequest:[NSURLRequest requestWithURL:[updateItem fileURL]] delegate:(id)self];
 }
 
 - (void)download:(NSURLDownload *)d decideDestinationWithSuggestedFilename:(NSString *)name
diff -u -r -x .svn -x .DS_Store -N Sparkle-1.5b6-OldPatches/SUHost.m Sparkle-1.5b6-NewPatches/SUHost.m
--- Sparkle-1.5b6-OldPatches/SUHost.m	2015-01-08 19:47:39.000000000 -0500
+++ Sparkle-1.5b6-NewPatches/SUHost.m	2015-01-15 13:18:43.000000000 -0500
@@ -20,7 +20,7 @@
 	{
         bundle = [aBundle retain];
 		if (![bundle bundleIdentifier])
-			NSLog(@"Sparkle Error: the bundle being updated at %@ has no CFBundleIdentifier! This will cause preference read/write to not work properly.");
+			NSLog(@"Sparkle Error: the bundle being updated at %@ has no CFBundleIdentifier! This will cause preference read/write to not work properly.", [aBundle bundlePath]);
     }
     return self;
 }
diff -u -r -x .svn -x .DS_Store -N Sparkle-1.5b6-OldPatches/SUPlainInstaller.m Sparkle-1.5b6-NewPatches/SUPlainInstaller.m
--- Sparkle-1.5b6-OldPatches/SUPlainInstaller.m	2014-10-28 19:44:46.000000000 -0400
+++ Sparkle-1.5b6-NewPatches/SUPlainInstaller.m	2015-01-15 21:17:40.000000000 -0500
@@ -18,9 +18,10 @@
 
 + (void)installPath:(NSString *)path overHost:(SUHost *)bundle delegate:delegate versionComparator:(id <SUVersionComparison>)comparator
 {
-	NSError *error;
+	NSError *error = nil;
 	BOOL result = YES;
 
+    /* DISABLE THIS TEST so we can "downgrade" from a beta to a stable version
 	// Prevent malicious downgrades:
 	if ([comparator compareVersion:[bundle version] toVersion:[[NSBundle bundleWithPath:path] objectForInfoDictionaryKey:@"CFBundleVersion"]] == NSOrderedDescending)
 	{
@@ -28,7 +29,8 @@
 		result = NO;
 		error = [NSError errorWithDomain:SUSparkleErrorDomain code:SUDowngradeError userInfo:[NSDictionary dictionaryWithObject:errorMessage forKey:NSLocalizedDescriptionKey]];
 	}
-	
+	*/
+    
 	if (result)
 		result = [self copyPathWithAuthentication:path overPath:[bundle bundlePath] error:&error];
 	[self _finishInstallationWithResult:result host:bundle error:error delegate:delegate];
diff -u -r -x .svn -x .DS_Store -N Sparkle-1.5b6-OldPatches/SUUIBasedUpdateDriver.m Sparkle-1.5b6-NewPatches/SUUIBasedUpdateDriver.m
--- Sparkle-1.5b6-OldPatches/SUUIBasedUpdateDriver.m	2014-10-28 19:44:46.000000000 -0400
+++ Sparkle-1.5b6-NewPatches/SUUIBasedUpdateDriver.m	2015-01-15 13:05:21.000000000 -0500
@@ -146,7 +146,7 @@
 
 - (void)abortUpdateWithError:(NSError *)error
 {
-	NSAlert *alert = [NSAlert alertWithMessageText:SULocalizedString(@"Update Error!", nil) defaultButton:SULocalizedString(@"Cancel Update", nil) alternateButton:nil otherButton:nil informativeTextWithFormat:[error localizedDescription]];
+	NSAlert *alert = [NSAlert alertWithMessageText:SULocalizedString(@"Update Error!", nil) defaultButton:SULocalizedString(@"Cancel Update", nil) alternateButton:nil otherButton:nil informativeTextWithFormat: @"%@", [error localizedDescription]];
 	[self showModalAlert:alert];
 	[super abortUpdateWithError:error];
 }
