diff -u -r -x .svn -x .DS_Store Sparkle-1.5b6-ORIGINAL/Info.plist Sparkle-1.5b6-PATCHED/Info.plist
--- Sparkle-1.5b6-ORIGINAL/Info.plist	2013-06-07 17:50:26.000000000 -0400
+++ Sparkle-1.5b6-PATCHED/Info.plist	2014-08-30 18:23:19.000000000 -0400
@@ -11,7 +11,7 @@
 	<key>CFBundleIconFile</key>
 	<string></string>
 	<key>CFBundleIdentifier</key>
-	<string>org.andymatuschak.Sparkle</string>
+	<string>net.tunnelblick.tunnelblick</string>
 	<key>CFBundleInfoDictionaryVersion</key>
 	<string>6.0</string>
 	<key>CFBundlePackageType</key>
diff -u -r -x .svn -x .DS_Store Sparkle-1.5b6-ORIGINAL/Sparkle.pch Sparkle-1.5b6-PATCHED/Sparkle.pch
--- Sparkle-1.5b6-ORIGINAL/Sparkle.pch	2013-06-07 17:50:26.000000000 -0400
+++ Sparkle-1.5b6-PATCHED/Sparkle.pch	2014-08-30 18:23:19.000000000 -0400
@@ -6,7 +6,7 @@
 //  Copyright 2008 Andy Matuschak. All rights reserved.
 //
 
-#define SPARKLE_BUNDLE [NSBundle bundleWithIdentifier:@"org.andymatuschak.Sparkle"]
+#define SPARKLE_BUNDLE [NSBundle bundleWithIdentifier:@"net.tunnelblick.tunnelblick"]
 #define SULocalizedString(key,comment) NSLocalizedStringFromTableInBundle(key, @"Sparkle", SPARKLE_BUNDLE, comment)
 #define SUAbstractFail() NSAssert2(nil, @"Can't call %@ on an instance of %@; this is an abstract method!", __PRETTY_FUNCTION__, [self class]);
 
