--- Sparkle/Sparkle/SUUpdater.m	2016-09-04 08:25:24.000000000 -0400
+++ Sparkle_/Sparkle/SUUpdater.m	2017-10-05 09:44:50.000000000 -0400
@@ -43,6 +43,8 @@
 @property (strong) SUUpdateDriver *driver;
 @property (strong) SUHost *host;
 
+@property (copy) NSDate *updateLastCheckedDate;
+
 @end
 
 @implementation SUUpdater
@@ -55,6 +57,7 @@
 @synthesize host;
 @synthesize sparkleBundle;
 @synthesize decryptionPassword;
+@synthesize updateLastCheckedDate;
 
 static NSMutableDictionary *sharedUpdaters = nil;
 static NSString *const SUUpdaterDefaultsObservationContext = @"SUUpdaterDefaultsObservationContext";
@@ -227,13 +230,19 @@
 
 - (NSDate *)lastUpdateCheckDate
 {
-    return [self.host objectForUserDefaultsKey:SULastCheckTimeKey];
+	if (![self updateLastCheckedDate])
+	{
+       [self setUpdateLastCheckedDate:[self.host objectForUserDefaultsKey:SULastCheckTimeKey]];
+	}
+	
+	return [self updateLastCheckedDate];
 }
 
 - (void)updateLastUpdateCheckDate
 {
     [self willChangeValueForKey:@"lastUpdateCheckDate"];
-    [self.host setObject:[NSDate date] forUserDefaultsKey:SULastCheckTimeKey];
+    [self setUpdateLastCheckedDate:[NSDate date]];
+    [self.host setObject:[self updateLastCheckedDate] forUserDefaultsKey:SULastCheckTimeKey];
     [self didChangeValueForKey:@"lastUpdateCheckDate"];
 }
 
