diff -u -r -x .svn -x .DS_Store -N Sparkle-1.5b6-OldPatches/SUBasicUpdateDriver.m Sparkle-1.5b6-NewPatches/SUBasicUpdateDriver.m
--- Sparkle-1.5b6/SUAppcast.m	2008-09-13 23:55:00.000000000 -0400
+++ Sparkle-1.5b6-Patched/SUAppcast.m	2016-01-30 15:08:16.000000000 -0500
@@ -50,7 +50,18 @@
 	CFRelease(connection);
     
 	NSError *error = nil;
-    NSXMLDocument *document = [[NSXMLDocument alloc] initWithData:incrementalData options:0 error:&error];
+	NSUInteger options = 0;
+	if (  NSAppKitVersionNumber <  1138 ) { // NSAppKitVersionNumber10_7 = 1138
+		// In order to avoid including external entities when parsing the appcast (a potential security vulnerability;
+		// see https://github.com/andymatuschak/Sparkle/issues/169), we ask NSXMLDocument to "tidy" the XML first.
+		// This happens to remove these external entities; it wouldn't be a future-proof approach, but it worked in these
+		// historical versions of OS X, and we have a more rigorous approach for 10.7+.
+		options = NSXMLDocumentTidyXML;
+	} else {
+		// In 10.7 and later, there's a real option for the behavior we desire.
+		options = (1UL << 19); // NSXMLNodeLoadExternalEntitiesNever is not defined in the 10.6 SDK
+	}
+    NSXMLDocument *document = [[NSXMLDocument alloc] initWithData:incrementalData options:options error:&error];
 	BOOL failed = NO;
 	NSArray *xmlItems = nil;
 	NSMutableArray *appcastItems = [NSMutableArray array];
