--- Sparkle/Configurations/ConfigCommon.xcconfig	2016-09-07 16:54:16.000000000 -0400
+++ Sparkle_/Configurations/ConfigCommon.xcconfig	2018-09-29 08:07:25.000000000 -0400
@@ -1,7 +1,7 @@
 // Common

 SPARKLE_BUNDLE_IDENTIFIER = org.sparkle-project.Sparkle
-SPARKLE_RELAUNCH_TOOL_NAME = Autoupdate
+SPARKLE_RELAUNCH_TOOL_NAME = TunnelblickUpdater
 SPARKLE_FILEOP_TOOL_NAME = fileop

 SPARKLE_APPEND_VERSION_NUMBER = 1
