diff -u -r -x .DS_Store openvpn-2.5_git_19d6d9c.old/src/openvpn/forward.c openvpn-2.5_git_19d6d9c.new/src/openvpn/forward.c
--- openvpn-2.5_git_19d6d9c.old/src/openvpn/forward.c	2019-02-27 07:06:12.000000000 -0500
+++ openvpn-2.5_git_19d6d9c.new/src/openvpn/forward.c	2019-02-27 07:06:12.000000000 -0500
@@ -947,7 +947,10 @@
 
     status = link_socket_read(c->c2.link_socket,
                               &c->c2.buf,
-                              &c->c2.from);
+			                  &c->c2.from,
+			                  c->options.ce.xormethod,
+			                  c->options.ce.xormask,
+			                  c->options.ce.xormasklen);
 
     if (socket_connection_reset(c->c2.link_socket, status))
     {
@@ -1588,7 +1591,10 @@
                 /* Send packet */
                 size = link_socket_write(c->c2.link_socket,
                                          &c->c2.to_link,
-                                         to_addr);
+                                         to_addr,
+				                         c->options.ce.xormethod,
+				                         c->options.ce.xormask,
+				                         c->options.ce.xormasklen);
 
                 /* Undo effect of prepend */
                 link_socket_write_post_size_adjust(&size, size_delta, &c->c2.to_link);
