diff -u -r -x .DS_Store openvpn-2.6.4.old/src/openvpn/forward.c openvpn-2.6.4.new/src/openvpn/forward.c
--- openvpn-2.6.4.old/src/openvpn/forward.c	2023-06-03 12:09:13.000000000 -0400
+++ openvpn-2.6.4.new/src/openvpn/forward.c	2023-06-03 12:09:13.000000000 -0400
@@ -936,7 +936,12 @@
     c->c2.buf = c->c2.buffers->read_link_buf;
     ASSERT(buf_init(&c->c2.buf, c->c2.frame.buf.headroom));

-    status = link_socket_read(sock, &c->c2.buf, &c->c2.from);
+    status = link_socket_read(sock,
+                              &c->c2.buf,
+                              &c->c2.from,
+                              c->options.ce.xormethod,
+                              c->options.ce.xormask,
+                              c->options.ce.xormasklen);

     if (socket_connection_reset(sock, status))
     {
@@ -1806,7 +1811,12 @@
                 socks_preprocess_outgoing_link(c, sock, &to_addr, &size_delta);

                 /* Send packet */
-                size = (int)link_socket_write(sock, &c->c2.to_link, to_addr);
+                size = (int)link_socket_write(sock,
+                                              &c->c2.to_link,
+                                              to_addr,
+                                              c->options.ce.xormethod,
+                                              c->options.ce.xormask,
+                                              c->options.ce.xormasklen);

                 /* Undo effect of prepend */
                 link_socket_write_post_size_adjust(&size, size_delta, &c->c2.to_link);
