diff -u -r openvpn-2.3.6/version.m4 openvpn-2.3.6_/version.m4
--- openvpn-2.3.6/version.m4	2014-11-29 10:00:35.000000000 -0500
+++ openvpn-2.3.6_/version.m4	2015-04-08 11:14:15.000000000 -0400
@@ -1,7 +1,7 @@
 dnl define the OpenVPN version
 define([PRODUCT_NAME], [OpenVPN])
 define([PRODUCT_TARNAME], [openvpn])
-define([PRODUCT_VERSION], [2.3.6])
+define([PRODUCT_VERSION], [2.3.6txp])
 define([PRODUCT_BUGREPORT], [openvpn-users@lists.sourceforge.net])
 define([PRODUCT_VERSION_RESOURCE], [2,3,6,0])
 dnl define the TAP version
