--- openvpn/src/openvpn/tun.c	2014-03-23 14:11:18.000000000 -0400
+++ patched/src/openvpn/tun.c	2014-03-27 15:18:22.000000000 -0400
@@ -30,6 +30,26 @@
  * from VTun by Maxim Krasnyansky <max_mk@yahoo.com>.
  */
 
+//*********************************************************************************************
+// BEGIN Tunnelblick patch
+// Tunnelblick is build on OS X 10.6.8 with Xcode 3.2.2 so it supports PPC processors.
+// However, the SDKs that come with that version of Xcode do not include if_utun.h
+// Rather than modify the build environment to add if_utun.h, we patch the OpenVPN source
+// code so that a straight OS X 10.6.8 with Xcode 3.2.2 will build Tunnelblick with utun support.
+//
+// We fool the source into thinking that if_utun.h is present (so that OpenVPN compiles utun
+// support) by defining "HAVE_NET_IF_UTUN_H".
+#define HAVE_NET_IF_UTUN_H 1
+//
+// We define the two variables that the OpenVPN source requires that are defined in if_utun.h,
+// "UTUN_CONTROL_NAME" and "UTUN_OPT_IFNAME". The values were taken from the OS X 10.7 file
+//    System/Library/Frameworks/Kernel.framework/Versions/A/Headers/net/if_utun.h
+// These values are the same in the SDKs for 10.6.8 (Xcode 3.2.6 only), 10.7, 10.8, and 10.9
+#define UTUN_CONTROL_NAME "com.apple.net.utun_control"
+#define UTUN_OPT_IFNAME   2
+// END   Tunnelblick patch
+//*********************************************************************************************
+
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #elif defined(_MSC_VER)
@@ -75,7 +95,14 @@
 
 #if defined(TARGET_DARWIN) && HAVE_NET_IF_UTUN_H
 #include <sys/kern_control.h>
-#include <net/if_utun.h>
+//*********************************************************************************************
+// BEGIN Tunnelblick patch
+// As described above, if_utun.h is NOT included in the SDK with which Tunnelblick is built,
+// and the definitions that this file uses from it are included above, so we comment out the
+// following "include" statement:
+// #include <net/if_utun.h>
+// END   Tunnelblick patch
+//*********************************************************************************************
 #include <sys/sys_domain.h>
 #endif
 
diff -u -r openvpn/src/openvpn/tun.h patched/src/openvpn/tun.h
--- openvpn/src/openvpn/tun.h	2014-03-23 14:11:18.000000000 -0400
+++ patched/src/openvpn/tun.h	2014-03-27 15:13:26.000000000 -0400
@@ -25,6 +25,19 @@
 #ifndef TUN_H
 #define TUN_H
 
+//*********************************************************************************************
+// BEGIN Tunnelblick patch
+// Tunnelblick is build on OS X 10.6.8 with Xcode 3.2.2 so it supports PPC processors.
+// However, the SDKs that come with that version of Xcode do not include if_utun.h
+// Rather than modify the build environment to add if_utun.h, we patch the OpenVPN source
+// code so that a straight OS X 10.6.8 with Xcode 3.2.2 will build Tunnelblick with utun support.
+//
+// We fool the source into thinking that if_utun.h is present (so that OpenVPN compiles utun
+// support) by defining "HAVE_NET_IF_UTUN_H"
+#define HAVE_NET_IF_UTUN_H 1
+// END   Tunnelblick patch
+//*********************************************************************************************
+
 #ifdef WIN32
 #include <winioctl.h>
 #include <tap-windows.h>
