diff -u -r -x .DS_Store openvpn-2.3.18.old/src/openvpn/forward.c openvpn-2.3.18.new/src/openvpn/forward.c
--- openvpn-2.3.18.old/src/openvpn/forward.c	2018-07-28 06:02:22.000000000 -0400
+++ openvpn-2.3.18.new/src/openvpn/forward.c	2018-07-28 06:02:23.000000000 -0400
@@ -674,7 +674,10 @@
 
   status = link_socket_read (c->c2.link_socket,
 			     &c->c2.buf,
-			     &c->c2.from);
+			     &c->c2.from,
+			     c->options.ce.xormethod,
+			     c->options.ce.xormask,
+			     c->options.ce.xormasklen);
 
   if (socket_connection_reset (c->c2.link_socket, status))
     {
@@ -1223,7 +1226,10 @@
 	    /* Send packet */
 	    size = link_socket_write (c->c2.link_socket,
 				      &c->c2.to_link,
-				      to_addr);
+				      to_addr,
+				      c->options.ce.xormethod,
+				      c->options.ce.xormask,
+				      c->options.ce.xormasklen);
 
 #ifdef ENABLE_SOCKS
 	    /* Undo effect of prepend */
