diff -u -r -x .DS_Store openvpn-2.3.18.old/src/openvpn/options.c openvpn-2.3.18.new/src/openvpn/options.c
--- openvpn-2.3.18.old/src/openvpn/options.c	2018-07-28 06:02:23.000000000 -0400
+++ openvpn-2.3.18.new/src/openvpn/options.c	2018-07-28 06:02:23.000000000 -0400
@@ -794,6 +794,9 @@
   o->max_routes = MAX_ROUTES_DEFAULT;
   o->resolve_retry_seconds = RESOLV_RETRY_INFINITE;
   o->proto_force = -1;
+  o->ce.xormethod = 0;
+  o->ce.xormask = "\0";
+  o->ce.xormasklen = 0;
 #ifdef ENABLE_OCC
   o->occ = true;
 #endif
@@ -911,6 +914,10 @@
   setenv_int_i (es, "local_port", e->local_port, i);
   setenv_str_i (es, "remote", e->remote, i);
   setenv_int_i (es, "remote_port", e->remote_port, i);
+  setenv_int_i (es, "xormethod", e->xormethod, i);
+  setenv_str_i (es, "xormask", e->xormask, i);
+  setenv_int_i (es, "xormasklen", e->xormasklen, i);
+
 
 #ifdef ENABLE_HTTP_PROXY
   if (e->http_proxy_options)
@@ -1370,6 +1377,9 @@
   SHOW_INT (connect_retry_seconds);
   SHOW_INT (connect_timeout);
   SHOW_INT (connect_retry_max);
+  SHOW_INT (xormethod);
+  SHOW_STR (xormask);
+  SHOW_INT (xormasklen);
 
 #ifdef ENABLE_HTTP_PROXY
   if (o->http_proxy_options)
@@ -5143,6 +5153,46 @@
       options->proto_force = proto_force;
       options->force_connection_list = true;
     }
+  else if (streq (p[0], "scramble") && p[1])
+    {
+      VERIFY_PERMISSION (OPT_P_GENERAL|OPT_P_CONNECTION);
+      if (streq (p[1], "xormask") && p[2] && (!p[3]))
+	{
+	  options->ce.xormethod = 1;
+	  options->ce.xormask = p[2];
+	  options->ce.xormasklen = strlen(options->ce.xormask);
+	}
+      else if (streq (p[1], "xorptrpos") && (!p[2]))
+	{
+	  options->ce.xormethod = 2;
+	  options->ce.xormask = NULL;
+	  options->ce.xormasklen = 0;
+	}
+      else if (streq (p[1], "reverse") && (!p[2]))
+	{
+	  options->ce.xormethod = 3;
+	  options->ce.xormask = NULL;
+	  options->ce.xormasklen = 0;
+	}
+      else if (streq (p[1], "obfuscate") && p[2] && (!p[3]))
+	{
+	  options->ce.xormethod = 4;
+	  options->ce.xormask = p[2];
+	  options->ce.xormasklen = strlen(options->ce.xormask);
+	}
+      else if (!p[2])
+	{
+	  msg (M_WARN, "WARNING: No recognized 'scramble' method specified; using 'scramble xormask \"%s\"'", p[1]);
+	  options->ce.xormethod = 1;
+	  options->ce.xormask = p[1];
+	  options->ce.xormasklen = strlen(options->ce.xormask);
+	}
+      else
+	{
+	  msg (msglevel, "No recognized 'scramble' method specified or extra parameters for 'scramble'");
+	  goto err;
+	}
+    }
 #ifdef ENABLE_HTTP_PROXY
   else if (streq (p[0], "http-proxy") && p[1])
     {
