diff -u -r original/src/openvpn/options.h patched/src/openvpn/options.h
--- original/src/openvpn/options.h	2016-08-23 10:16:22.000000000 -0400
+++ patched/src/openvpn/options.h	2016-08-24 08:19:08.000000000 -0400
@@ -100,6 +100,9 @@
   int connect_retry_max;
   int connect_timeout;
   bool connect_timeout_defined;
+  int xormethod;
+  const char *xormask;
+  int xormasklen;
 #ifdef ENABLE_HTTP_PROXY
   struct http_proxy_options *http_proxy_options;
 #endif  
