diff -u -r openvpn-2.3.7/version.m4 openvpn-2.3.7_/version.m4
--- openvpn-2.3.7/version.m4
+++ openvpn-2.3.7_/version.m4
@@ -1,7 +1,7 @@
 dnl define the OpenVPN version
 define([PRODUCT_NAME], [OpenVPN])
 define([PRODUCT_TARNAME], [openvpn])
-define([PRODUCT_VERSION], [2.3.7])
+define([PRODUCT_VERSION], [2.3.7txp])
 define([PRODUCT_BUGREPORT], [openvpn-users@lists.sourceforge.net])
 define([PRODUCT_VERSION_RESOURCE], [2,3,7,0])
 dnl define the TAP version
